### PR TITLE
test: added tests for /Mutation/updateAdvertisement.ts

### PR DESCRIPTION
**Pull request by MonishPuttu**
Fixes: #3832

### Authentication & Authorization (4 tests)
- Unauthenticated user error
- Deleted user with valid token (edge case)
- Non-admin organization member unauthorized
- Regular user with no organization membership unauthorized

### Input Validation (1 test)
- Invalid UUID format for advertisement ID

### Resource Validation (1 test)
- Non-existent advertisement error

### Business Logic (3 tests)
- Invalid startAt validation (new startAt after existing endAt)
- Invalid endAt validation (new endAt before existing startAt)
- Duplicate advertisement name within same organization

### Database Error Handling (1 test)
- Database update operation failure (transaction returns empty array)

### Success Scenarios (5 tests)
- Successfully update advertisement name only
- Successfully update advertisement description
- Successfully update advertisement type (banner to pop_up)
- Successfully update both startAt and endAt dates
- Successfully update multiple fields simultaneously (name, description, type)

## Test Coverage

**Statement** **100%**  
**Branch** **100%**  
**Function** **100%**

## Files Modified
- `test/graphql/types/Mutation/updateAdvertisement.test.ts` - Complete integration test suite (15 tests)

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevented false duplicate-name errors when updating an advertisement’s name by excluding the record being updated from the uniqueness check.

* **Tests**
  * Added comprehensive tests for updateAdvertisement covering auth, input validation, permissions, error handling, and successful update scenarios.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by MonishPuttu** [comment]
@coderabbitai review

**Comment by MonishPuttu** [comment]
@coderabbitai 
Uncovered Line Note: Line 188 represents a race condition defense (record deleted between validation and update). Error handling is tested via mock in the "unexpected database failure" test. 99.53% coverage comprehensively covers all user-facing code paths and reflects realistic test scenarios.
Is there any alternative fix?

File: test/graphql/types/Mutation/updateAdvertisement.test.ts [hunk 1]
```diff
@@ -0,0 +1,1068 @@
...
+					},
+				},
+			);
+
+			expect(result.errors).toBeUndefined();
+			expect(result.data?.updateAdvertisement).toEqual(
+				expect.objectContaining({
+					id: adId,
+				}),
+			);
```
[the remark this answers is not in the document]
Comment (MonishPuttu) [reply test/graphql/types/Mutation/updateAdvertisement.test.ts:1055]: working on it

**Comment by MonishPuttu** [comment]
@coderabbitai completed the changes mentioned, can you review?

**Review by palisadoes — APPROVED** [review]
